.circleci/config.yml

```diff
@@ -45,7 +45,7 @@ jobs:
           name: Run unit tests
           command: |
             cd << parameters.project_dir >>
-            clojure -X:test:test/unit
+            clojure -X:dev:test:test/unit
```

Comment:
What's in the`dev` profile that is required for testing? If tests require some random stuff from `dev`, then we should move that to our `test` folder (or stop relying on it).

Reply:
Short annoying answer: because that how we do things here!

Slightly longer answer...

Running `:dev` alias with additional deps and config for `:test` is the usual way that we run things in most of our projects. For example see muttnik here: https://github.com/Swirrl/muttnik/blob/c874743132467fa594d18b5b4ef7c0638b47f416/bin/cider#L5

If you start a REPL running only with the :dev alias but want to subsequently run, edit and debug a test in that same REPL, it's easier to just use `:dev:test`. That's how I like to develop. Others do too. I don't really want to start another REPL or restart the one I'm currently working in.

I didn't notice the change in behaviour until I noticed a dupe dep had been added to `:test` that already existed in `:dev`. That is what the following commit addresses:

https://github.com/Swirrl/datahost-prototypes/pull/320/commits/ecb9443091217e725b3f37684006ac705b7d401a

Essentially, I have put things back to how they've always been. Which is that `test` alias is an _extended_ version of what's in `dev`.

This is how it's been since the start of the project.

Here is the README from May 2023, before the example start command was moved from the README into Kaocha:

https://github.com/Swirrl/datahost-prototypes/commit/b04c6d6642ee1f578f507e950dce80da582b2787

And here is the Kaocha command:

https://github.com/Swirrl/datahost-prototypes/blob/20b16b21d1c41269ce2da0976b379b30304a466e/datahost-ld-openapi/bin/kaocha

Reply:
That's all fine when a developer runs the project on their machine (e.g. I pretty much always include dev and test profiles locally),  but this is CI - anything that's not the 'app' code or 'test' code is an unknown and that's why I think it should not be included. Worst case scenario: a test run with `:dev` profile can be successful, but it's possible the app won't even start without it (or behaves slightly differently, because it picked up different config etc).

Reply:
Offending alias commits have been nuked. 💥